pvlib/pvsystem.py

```diff
@@ -836,16 +837,24 @@ def physicaliam(aoi, n=1.526, K=4., L=0.002):
     # after deducting the reflected portion of each
     iam = ((1 - (rho_para + rho_perp) / 2) / (1 - rho_zero) * tau / tau_zero)
 
+    # set up array for nan-tolerant masking operations
+    notnan = ~np.isnan(aoi)
+
     # angles near zero produce nan, but iam is defined as one
     small_angle = 1e-06
-    iam = np.where(np.abs(aoi) < small_angle, 1.0, iam)
+    aoi_lt_small_angle = np.full_like(aoi, False, dtype='bool')
+    np.less(np.abs(aoi), small_angle, where=notnan, out=aoi_lt_small_angle)
+    iam = np.where(aoi_lt_small_angle, 1.0, iam)
 
-    # angles at 90 degrees can produce tiny negative values, which should be zero
-    # this is a result of calculation precision rather than the physical model
-    iam = np.where(iam < 0, 0, iam)
+    # angles at 90 degrees can produce tiny negative values,
+    # which should be zero. this is a result of calculation precision
+    # rather than the physical model
+    iam = np.maximum(0, iam)
 
     # for light coming from behind the plane, none can enter the module
-    iam = np.where(aoi > 90, 0, iam)
+    aoi_gt_90 = np.full_like(aoi, False, dtype='bool')
+    np.greater(aoi, 90, where=notnan, out=aoi_gt_90)
+    iam = np.where(aoi_gt_90, 0, iam)
```

Comment:
I would suggest a block ignore invalid on these (originally) three statements.  I think aoi nans propagate fine to iam, and then the logic is all correct the way it is.

Reply:
I agree that the errstate call is easier to read here and I don't think we lose anything by using it here. Updated accordingly.